src/realm/query_conditions_tpl.hpp

```diff
@@ -33,8 +33,27 @@ class QueryStateSum : public QueryStateBase {
     using Type = T;
     using ResultType = typename aggregate_operations::Sum<T>::ResultType;
     using QueryStateBase::QueryStateBase;
-    bool match(size_t, Mixed value) noexcept final
+    bool match(size_t index, Mixed value) noexcept final
+    {
+        if (m_source_column) {
+            REALM_ASSERT_DEBUG(value.is_null());
+            value = m_source_column->get_any(index);
+        }
+        else {
+            static_cast<void>(m_source_column);
```

Comment:
Shouldn't be any reason to need this. If it was an `if constexpr` guarding the use of m_source_column you might get some spurious warnings but here the member variable is always used.